src/functions/src/main/java/org/apache/jmeter/functions/TimeFunction.java

```diff
@@ -96,13 +99,17 @@ public String execute(SampleResult previousResult, Sampler currentSampler) throw
                 long div = Long.parseLong(fmt.substring(1)); // should never case NFE
                 datetime = Long.toString(System.currentTimeMillis() / div);
             } else {
-                if (fmt.contains("u")) {
+                DateTimeFormatter df = DateTimeFormatter
+                        .ofPattern(fmt);
```

Comment:
We could probably move the initialization of `DateTimeFormatter` to something like "once for a given pattern" since `DateTimeFormatter` is thread-safe.

That might be a task for another PR though.